internal/service/certificatesigningrequest.go

```diff
@@ -243,6 +240,13 @@ func (h *ServiceHandler) PatchCertificateSigningRequest(ctx context.Context, req
 func (h *ServiceHandler) ReplaceCertificateSigningRequest(ctx context.Context, request server.ReplaceCertificateSigningRequestRequestObject) (server.ReplaceCertificateSigningRequestResponseObject, error) {
 	orgId := store.NullOrgId
 
+	// check that the CSR does not already exist
+	_, err := h.store.CertificateSigningRequest().Get(ctx, orgId, request.Name)
+	if err != flterrors.ErrResourceNotFound {
```

Comment:
technically nil is also true here right?
```suggestion
	if err != nil {
	  if !errors.Is(flterrors.ErrResourceNotFound) {
```

Reply:
In this case we want a nil error to evaluate to true because both a nil error and a non-ResourceNotFound error both mean the resource exists. ErrResourceNotFound is the only acceptable result of CertificateSigningRequest().Get(...) to execute the rest of the logic.

Reply:
Ok then I would still recommend using errors.Is so that the error in the future could be `%w` wrapped and this eval would fail. Your call on that but it's a good practice in general.
```
if !errors.Is(err, flterrors.ErrResourceNotFound)
```

Reply:
Good suggestion, updated. I'm still getting used to this go syntax.